packages/tailwindcss/src/utilities.ts

```diff
@@ -2261,122 +2261,205 @@ export function createUtilities(theme: Theme) {
+            }
+            default: {
+              if (negative) return
+
+              return [
+                decl('--tw-gradient-position', `${value},`),
+                decl('background-image', `linear-gradient(var(--tw-gradient-stops,${value}))`),
+              ]
+            }
           }
         }
-      } else {
-        if (!isPositiveInteger(value)) return
 
-        value = negative ? `calc(${value}deg * -1)` : `${value}deg`
+        let value = candidate.value.value
+
+        if (!negative && linearGradientDirections.has(value)) {
+          value = linearGradientDirections.get(value)!
+        } else if (isPositiveInteger(value)) {
+          value = negative ? `calc(${value}deg * -1)` : `${value}deg`
+        } else {
+          return
+        }
+
+        let interpolationMethod = resolveInterpolationModifier(candidate.modifier)
 
         return [
-          decl('--tw-gradient-position', `${value} in oklch,`),
+          decl('--tw-gradient-position', `${value} ${interpolationMethod},`),
           decl('background-image', `linear-gradient(var(--tw-gradient-stops))`),
         ]
       }
     }
-  }
 
-  utilities.functional('-bg-linear', handleBgLinear({ negative: true }))
-  utilities.functional('bg-linear', handleBgLinear({ negative: false }))
+    utilities.functional('-bg-linear', handleBgLinear({ negative: true }))
+    utilities.functional('bg-linear', handleBgLinear({ negative: false }))
 
-  function handleBgConic({ negative }: { negative: boolean }) {
-    return (candidate: Extract<Candidate, { kind: 'functional' }>) => {
-      if (candidate.modifier) return
+    suggest('bg-linear', () => [
+      {
+        values: [...linearGradientDirections.keys()],
+        modifiers: suggestedModifiers,
+      },
+      {
+        values: ['0', '30', '60', '90', '120', '150', '180', '210', '240', '270', '300', '330'],
+        supportsNegative: true,
+        modifiers: suggestedModifiers,
+      },
+    ])
 
-      if (!candidate.value) {
-        return [
-          decl('--tw-gradient-position', `in oklch,`),
-          decl('background-image', `conic-gradient(var(--tw-gradient-stops))`),
-        ]
-      }
+    function handleBgConic({ negative }: { negative: boolean }) {
+      return (candidate: Extract<Candidate, { kind: 'functional' }>) => {
+        if (candidate.value?.kind === 'arbitrary') {
+          if (candidate.modifier) return
+          let value = candidate.value.value
+          return [
+            decl('--tw-gradient-position', `${value},`),
+            decl('background-image', `conic-gradient(var(--tw-gradient-stops,${value}))`),
+          ]
+        }
 
-      let value = candidate.value.value
+        let interpolationMethod = resolveInterpolationModifier(candidate.modifier)
+
+        if (!candidate.value) {
+          return [
+            decl('--tw-gradient-position', `${interpolationMethod},`),
+            decl('background-image', `conic-gradient(var(--tw-gradient-stops))`),
```

Comment:
The arbitrary value has a fallback here, should we add one here too (or rm the fallback above)?

Reply:
This is for the `bg-conic` utility with no value, which just can't do anything at all unless you also use the `from-*` and `to-*` utilities, so since it's not possible for the utility to have any effect on its own it didn't make sense to me to include the fallback.

This is just junk CSS for example:

```
background-image: conic-gradient(in oklch);
```

…and that's what you'd be getting if the fallback triggered. Since it's invalid either way it's less code to just not include it, and also means we don't imply that it's supposed to work or anything.

Reply:
ah yes makes sense, I was confused there because I didn't read the variables used correctly...